ocs_ci/ocs/node.py

```diff
@@ -908,6 +914,105 @@ def delete_and_create_osd_node_vsphere_upi(osd_node_name, use_existing_node=Fals
     return new_node_name
 
 
+def delete_and_create_osd_node_vsphere_lso(osd_node_name, use_existing_node=False):
```

Comment:
can this function changed to be more generic, to delete and create osd on any platform and for both LSO and dynamically provisioned clusters?

Reply:
The steps are different. In the LSO node, we need to perform additional steps. Maybe we can move it to a class like the `platform_nodes` file, but it will take some time. Anyway, if you have any idea how to make it more generic please let me know.

Reply:
adding it to platform_nodes is an option, although, platform_nodes should be platform oriented and not OCP oriented. 

Another option would be to create a new class structure for OCP node operations, which can include node replacement, node tain, etc. This new class structure can will have child classed for each platform / deployment method combination. Something like:

Reply:
I think it will be better to merge this PR and then implement a new PR that will handle the new functionality. 
I created a new issue for that https://github.com/red-hat-storage/ocs-ci/issues/3648.

Reply:
ack